runtime/src/fips.rs

```diff
@@ -37,55 +36,122 @@ impl FipsModule {
         }
         env.persistent_data.get_mut().zeroize();
     }
+}
 
-    /// Execute KAT for cryptographic algorithms implemented in H/W.
-    fn execute_kats(env: &mut Drivers) -> CaliptraResult<()> {
-        cprintln!("[kat] Executing SHA2-256 Engine KAT");
-        Sha256Kat::default().execute(&mut env.sha256)?;
-
-        cprintln!("[kat] Executing SHA2-384 Engine KAT");
-        Sha384Kat::default().execute(&mut env.sha384)?;
-
-        cprintln!("[kat] Executing SHA2-384 Accelerator KAT");
-        Sha384AccKat::default().execute(&mut env.sha384_acc)?;
+#[cfg(feature = "fips_self_test")]
+pub mod fips_self_test_cmd {
+    use super::*;
+    use crate::RtBootStatus::{RtFipSelfTestComplete, RtFipSelfTestStarted};
+    use caliptra_common::{verifier::FirmwareImageVerificationEnv, FMC_ORG, RUNTIME_ORG};
+    use caliptra_drivers::ResetReason;
+    use caliptra_image_verify::ImageVerifier;
+    use zerocopy::AsBytes;
+
+    // Helper function to create a mutable slice from a memory region
+    unsafe fn create_slice(org: u32, size: usize) -> &'static mut [u8] {
+        let ptr = org as *mut u8;
+        core::slice::from_raw_parts_mut(ptr, size)
+    }
+    pub enum SelfTestStatus {
+        Idle,
+        InProgress(fn(&mut Drivers) -> CaliptraResult<()>),
+        Done,
+    }
 
-        cprintln!("[kat] Executing ECC-384 Engine KAT");
-        Ecc384Kat::default().execute(&mut env.ecc384, &mut env.trng)?;
+    fn copy_and_verify_image(env: &mut Drivers) -> CaliptraResult<()> {
+        cprintln!("write dummy cmd");
```

Comment:
Nit: these cprintln! statements aren't that informative. Not a blocker for merge.

Reply:
Removed.